agent/connect/ca/provider_vault.go

```diff
@@ -101,9 +101,15 @@ func (v *VaultProvider) GenerateRoot() error {
 		if err != nil {
 			return err
 		}
+		keyType := v.config.PrivateKeyType
+		if keyType == "ecdsa" {
+			keyType = "ec"
```

Comment:
Should we just use `ec` rather than `ecdsa` for our key type string?

Arguments for:
 - it's more correct, it _is_ and "Eliptic Curve" key which can be _used for_ Eliptic Curve Digital Signature Algorim signing"
 - it's concistent with Vault
 - it's more consistent with PEM (`EC PRIVATE KEY` vs `RSA PRIVATE KEY` block types)

Any reason to use `ecdsa` I'm missing?

Reply:
I'm fine with either one. I used `ecdsa` because that's the name of the package (`ecdsa.PrivateKey`).

Reply:
Fixed with 876b9caa43f348460f17735e6ca75474911368ba.